bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java

```diff
@@ -109,9 +114,24 @@ public Client(@NonNull Context androidContext, @NonNull String apiKey) {
      * @param configuration  a configuration for the Client
      */
     public Client(@NonNull Context androidContext, @NonNull final Configuration configuration) {
+        // if the user has set the releaseStage to production manually, disable logging
+        if (configuration.getLogger() == null) {
+            String releaseStage = configuration.getReleaseStage();
+            boolean loggingEnabled = !AppData.RELEASE_STAGE_PRODUCTION.equals(releaseStage);
+
+            if (loggingEnabled) {
+                configuration.setLogger(DebugLogger.INSTANCE);
```

Comment:
Why do you mutate configuration here? I guess it's not a deal breaker for this PR but it just doesn't make sense in my head to make changes after it has been supplied.

Reply:
The object is converted to an object of type ImmutableConfiguration, and setting a logger on the original object was the simplest way to pass existing sanitisation code.